tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -138,14 +174,90 @@ func (t *testConfig) Setup() (err error) {
 	return
 }
 
-func createDefaultRoutingRules() error {
-	for _, rule := range []string{bookinfoGateway, routeAllRule} {
-		if err := createRouteRule(rule); err != nil {
-			return fmt.Errorf("could not create base routing rules: %v", err)
+func copyRuleToFilesystem(t *testConfig, rule string) error {
+	src := getSourceRulePath(rule)
+	dest := getDestinationRulePath(t, rule)
+	log.Infof("Copying rule %s from %s to %s", rule, src, dest)
+	ori, err := ioutil.ReadFile(src)
+	if err != nil {
+		log.Errorf("Failed to read original rule file %s", src)
+		return err
+	}
+	content := string(ori)
+	// TODO: is this needed and why?
```

Comment:
This is needed for the tests that involve routing per user. Instead of logging in as the user `jason` of the bookinfo rules, to perform the tests, log in of the user specified by the `u2` variable is performed. To achieve this routing per `u2` user, the string `jason` is replaced in the route rules by the value of the variable `u2`.

Reply:
Then I think for mixer tests it can be eliminated.
Can this var be localized to that test? Seeing it at top level is confusing.

Reply:
This var is global in order to perform this transformation - replacing `jason` user with the value of `u2` variables, for all the rules and to use the same value in the tests. And the assumption here is that several tests may use it.